### PR TITLE
audi - C++ header only

### PR DESCRIPTION
**Pull request by darioizzo**
C++ library for autodiff and differential algebra.
I will then create a separate recipe for the python exposition of the package pyaudi

### CODE REVIEW

**Comment by darioizzo** [comment]
Not sure why CircleCi is failing here. Anyone can help?
The only message given is:

```
#!/bin/bash -eo pipefail
./scripts/run_docker_build.sh

/bin/bash: ./scripts/run_docker_build.sh: No such file or directory
Exited with code 1
```

**Comment by darioizzo** [comment]
@conda-forge-admin, please rerender

**Comment by darioizzo** [comment]
@isuruf Can you have a look? I am unable to proceed.

**Comment by isuruf** [comment]
Can you merge with master?

**Comment by darioizzo** [comment]
@isuruf The recipe is ready as far as I am concerned. Can be merged. Is that what you meant? Or I should do something?

**Comment by isuruf** [comment]
You need to merge your branch with conda-forge master for the circleci tests to work.

**Comment by darioizzo** [comment]
@isuruf Thanks! Somehow I had not noticed I was not even with staged-recipes master.

**Comment by darioizzo** [comment]
@isuruf Looks now all is in order. If you can merge this I can then create the conda recipe package for the python module pyaudi.

File: recipes/audi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+requirements:
+  build:
+    - toolchain
+    - cmake
+    - boost-cpp 1.64.0
+    - eigen
+    - piranha 0.10
+
+  run:
+    - boost-cpp 1.64.0
```
Comment (isuruf) [inline recipes/audi/meta.yaml]: Can you change this to 1.66.0?

Comment (darioizzo) [reply recipes/audi/meta.yaml]: piranha 0.10 is pinned to 1.64, unfortunately. @bluescarni is working on a code refactoring of piranha, in the meantime I have to use v0.10

Comment (isuruf) [reply recipes/audi/meta.yaml]: Is it possible to rebuild piranha 0.10 with 1.66?

Comment (darioizzo) [reply recipes/audi/meta.yaml]: I think it should be, @bluescarni any reason not to?

**Comment by darioizzo** [comment]
@isuruf Can this be merged now?

**Pull request merged**